Add Documentation for the CLI mode (#19)
wso2-extensions/identity-tools-cli: docs/interactive-mode.md

# Interactive Mode

> The interactive mode can be used to handle application configurations in an interactive manner. This can be used to add, list, export and import applications in the target environment.

Reviewer: ```suggestion
The interactive mode can be used to handle application configurations in an interactive manner. This can be used to add, list, export, and import applications in the target environment.
```

> Note: This mode does not provide support for bulk resource export or import.

> ### Running the tool in the interactive mode

Reviewer: ```suggestion
### Run the tool in the interactive mode

See the topics given below to run the tool in interactive mode.
```

> #### Tool Initialization

Reviewer: ```suggestion
#### Tool initialization
```

> 1. Setup the tool and the IS, following the steps in the [How to run the tool ](../README.md##How to run the tool ) section.

Reviewer: ```suggestion
1. Set up the tool and WSO2 IS following the steps in the [How to run the tool ](../README.md##How to run the tool ) section.
```

> 2. Run the following command to initialize the tool by providing the Identity server details and client ID/secret of the app you created.

Reviewer: ```suggestion
2. Run the following command to initialize the tool by providing details of WSO2 IS and the client ID/secret of the app you created.
```

  | |    / _ \   | |\/| |  _____  | |       | |   | |    
  | |   / ___ \  | |  | | |_____| | |___    | |   | |___ 
 |___| /_/   \_\ |_|  |_|          \____|   |_|   |_____|
      
? Enter IAM URL [<schema>://<host>]: https://localhost:9443                                                   
? Enter clientID: *******
? Enter clientSecret: *******
? Enter Tenant domain: carbon.super
```
Run the following command to provide admin user credentials.
```
iamctl serverConfiguration [flags]
```

Flags:
```
  -h, --help              help for serverConfiguration
  -p, --password string   enter your password
  -s, --server string     set server domain
  -u, --username string   enter your username
```
example:-
```
iamctl serverConfiguration -h                                           //help for serverConfiguration
iamctl serverConfiguration -s=https://localhost:9443 -u=admin -p=*****  //to complete the authorization
```
Set admin user credentials by entering inputs in an interactive way.
```
iamctl serverConfiguration
```
example:-
```
~$ iamctl serverConfiguration 
? Enter IAM URL [<schema>://<host>]: https://localhost:9443
? Enter Username: admin
? Enter Password: *****
```

> ### Application Related Commands

Reviewer: ```suggestion
### Application-related commands
```

**Add application**
```
iamctl application [commands]
iamctl application     add      [flags]
```

Flags:
 ```
   -c, --callbackURl string    callbackURL  of SP - **for oauth application
   -d, --description string    description of SP - **for basic application
   -h, --help                  help for add
   -n, --name string           name of service provider - **compulsory
   -p, --password string       Password for Identity Server
   -s, --serverDomain string   server Domain
   -t, --type string           Enter application type (default "oauth")
   -u, --userName string       Username for Identity Server
 ```

> Users have freedom to set flags and values according to their choices.

Reviewer: ```suggestion
Users have the freedom to set flags and values according to their choices.
```

> This ```-t, --type string           Enter application type (default "oauth")``` flag  is not mandatory. If user wants to create basic application, then should declare ```-t=basic```. Otherwise will create the oauth application as default type.

Reviewer: ```suggestion
This ```-t, --type string           Enter application type (default "oauth")``` flag  is not mandatory. If the user wants to create a basic application, it is necessary to declare ```-t=basic```. Otherwise, the OAuth application will be created as the default type.
```

example:-
```
//create an oauth application
iamctl application add  -n=TestApplication 
iamctl application add -t=oauth -n=TestApplication
iamctl application add -t=oauth -n=TestApplication -d=description
iamctl application add -t=oauth -n=TestApplication -c=https://localhost:8010/oauth
iamctl application add -t=oauth -n=TestApplication -c=https://localhost:8010/oauth -d=description

//create an basic application
iamctl application add -t=basic -n=TestApplication
iamctl application add -t=basic -n=TestApplication -d=description
```

> You can set server domain and create application at the same time.

Reviewer: ```suggestion
You can set the server domain and create an application at the same time.
```

example:-
```
//create an oauth application
iamctl application add -s=https://localhost:9443 -u=admin -p=***** -n=TestApplication 
iamctl application add -s=https://localhost:9443 -u=admin -p=***** -t=oauth -n=TestApplication
iamctl application add -s=https://localhost:9443 -u=admin -p=***** -t=oauth -n=TestApplication -d=description
iamctl application add -s=https://localhost:9443 -u=admin -p=***** -t=oauth -n=TestApplication -c=https://localhost:8010/oauth
iamctl application add -s=https://localhost:9443 -u=admin -p=***** -t=oauth -n=TestApplication -c=https://localhost:8010/oauth -d=description

//create an basic application
iamctl application add -s=https://localhost:9443 -u=admin -p=***** -t=basic -n=TestApplication
iamctl application add -s=https://localhost:9443 -u=admin -p=***** -t=basic -n=TestApplication -d=description
```

> **Get list of applications**

Reviewer: ```suggestion
**Get a list of applications**
```

```
iamctl application     list     [flags]
```
Flags:
```
  -h, --help              help for list
  -p, --password string   Password for Identity Server
  -s, --server string     server
  -u, --userName string   User name for Identity Server
```
example:-
```
//get list of applications
iamctl application list 
```

> You cat set server domain and get the list of applications at the same time.

Reviewer: ```suggestion
You can set the server domain and get the list of applications at the same time.
```

example:-
```
//get list of applications
iamctl application list -s=https://localhost:9443 -u=admin -p=*****
```

> #### Create service providers by entering inputs in an interactive way.

Reviewer: ```suggestion
#### Create service providers by entering inputs in an interactive way
```

**Add application and get list of applications**

```
iamctl application
```

> It gives following output after entering the server domain.

Reviewer: ```suggestion
It gives the following output after entering the server domain.
```
